iDiff/README.md

```diff
@@ -0,0 +1,241 @@
+
+Docker history lines found only in gcr.io/google-appengine/python:2017-07-25-110644:
+-/bin/sh -c #(nop) ADD file:9b537477c1fe03a9a3af141199b8848b9718bbc259e9d040e52dd78d9b1472a0 in /
+-/bin/sh -c #(nop) ADD dir:b209be879a64da94090efa46c7647cf4a972d9233219a86718dea815b8b6ea62 in /scripts
+-/bin/sh -c #(nop) ADD dir:de273ffbfc6ea318de85b56ed05cd1d002b5b0bfa5c721a42b7aa8d44ff60c42 in /resources
+
+Packages found only in pythonlatest:
+
+Packages found only in python2017-07-25-110644:
+
+Version differences:
+	(Package:	pythonlatest		python2017-07-25-110644)
+
+Packages found only in pythonlatest:
+
+Packages found only in python2017-07-25-110644:
+
+Version differences:
+	(Package:	pythonlatest		python2017-07-25-110644)
+
+```
+
+
+## Make your own differ
+
+Feel free to develop your own differ leveraging the utils currently available.  PRs are welcome.
+
+### Custom Differ Quickstart
+
+In order to quickly make your own differ, follow these steps:
+
+1. Add your diff identifier to the flags in [root.go](https://github.com/GoogleCloudPlatform/runtimes-common/blob/ReadMe/iDiff/cmd/root.go)
+2. Determine if you can use existing differ tools.  If you can make use of existing tools, you then need to construct the structs to feed to the diff tools by getting all of the packages for each image or the analogous quality to be diffed.  To determine if you can leverage existing tools, think through these questions:
+- Are you trying to diff packages?
+    - Yes: Does the relevant package manager support different versions of the same package on one image?
+        - Yes: Use `GetMultiVerisonMapDiff` to diff `map[string]map[string]utils.PackageInfo` objects.  See [nodeDiff.go](https://github.com/GoogleCloudPlatform/runtimes-common/blob/master/iDiff/differs/nodeDiff.go#L33) for an example.
+        -  No: Use `GetMapDiff` to diff `map[string]utils.PackageInfo` objects.  See [aptDiff.go](https://github.com/GoogleCloudPlatform/runtimes-common/blob/master/iDiff/differs/aptDiff.go#L29) or [pipDiff.go](https://github.com/GoogleCloudPlatform/runtimes-common/blob/master/iDiff/differs/pipDiff.go#L23) for examples. 
+    - No: Look to [History](https://github.com/GoogleCloudPlatform/runtimes-common/blob/ReadMe/iDiff/differs/historyDiff.go) and [File System](https://github.com/GoogleCloudPlatform/runtimes-common/blob/ReadMe/iDiff/differs/fileDiff.go) differs as models for diffing.
+
+3. Write your Diff driver such that you have a struct for your differ type and a method for that differ called Diff:
+
+```
+type YourDiffer struct {}
+
+func (d YourDiffer) Diff(image1, image2 utils.Image) (DiffResult, error) {...}
+```
+The arguments passed to your differ contain the path to the unpacked tar representation of the image.  That path can be accessed as such: `image1.FSPath`.  Given that path you should create the appropriate struct (determined in step 2) and then call the appropriate get-diff function (also determined in step2).
```

Comment:
I think the most likely differ to be made by someone is a package differ so I hesitate to omit this part.  I tried to break it up a little more for package differ and not, as well as directly referencing the get-diff functions I was referring to. 

Do you have wording you would propose to make the relation between the image path and the next part more clear?